data_safe_haven/logging/plain_file_handler.py

```diff
@@ -22,7 +22,14 @@ def strip_formatting(input_string: str) -> str:
         text.spans = []
         return str(text)
 
+    @staticmethod
+    def convert_text(text: Text) -> str:
+        """Convert a rich.text.Text object to a string"""
+        return str(text)
+
     def emit(self, record: logging.LogRecord) -> None:
         """Emit a record without formatting"""
+        if isinstance(record.msg, Text):
+            record.msg = self.convert_text(record.msg)
```

Comment:
Why not do this inside the `from_ansi` function?

Reply:
I think we only want to do this for the `PlainFileHandler`.

The rich console handlers can deal with `Text` objects and should render the formatting.